### PR TITLE
media: i2c: imx477: Add vsync trigger_mode parameter

### PR DESCRIPTION
**Pull request by neocortex-vision**
trigger_mode == 0 (default) => no effect / no registers written
trigger_mode == 1           => source
trigger_mode == 2           => sink

This can be set e.g. in /boot/cmdline.txt as imx477.trigger_mode=N
Setting it to 0 or not setting it at all has the same effect.

Tested on HQ cameras (XVS <-> XVS, GND <-> GND) with libcamera.

Credits to 6by9.

### CODE REVIEW

**Comment by 6by9** [comment]
The content looks OK, although we get a couple of warnings from ```scripts/checkpatch.pl --strict --codespell```
```
No codespell typos will be found - file '/usr/share/codespell/dictionary.txt': No such file or directory
ERROR: do not initialise statics to 0
#25: FILE: drivers/media/i2c/imx477.c:28:
+static int trigger_mode = 0;

CHECK: Alignment should match open parenthesis
#55: FILE: drivers/media/i2c/imx477.c:1738:
+		imx477_write_reg(imx477, IMX477_REG_MC_MODE,
+			IMX477_REG_VALUE_08BIT, 1);

CHECK: Alignment should match open parenthesis
#57: FILE: drivers/media/i2c/imx477.c:1740:
+		imx477_write_reg(imx477, IMX477_REG_MS_SEL,
+			IMX477_REG_VALUE_08BIT, val);

CHECK: Alignment should match open parenthesis
#59: FILE: drivers/media/i2c/imx477.c:1742:
+		imx477_write_reg(imx477, IMX477_REG_XVS_IO_CTRL,
+			IMX477_REG_VALUE_08BIT, val);

CHECK: Alignment should match open parenthesis
#61: FILE: drivers/media/i2c/imx477.c:1744:
+		imx477_write_reg(imx477, IMX477_REG_EXTOUT_EN,
+			IMX477_REG_VALUE_08BIT, val);

ERROR: Missing Signed-off-by: line(s)

total: 2 errors, 0 warnings, 4 checks, 43 lines checked
```
As currently imx477.c is totally clean of checkpatch warnings, it would be nice to fix all of those up, even though ```--strict``` is not strictly required (those on linux-media seem to expect it though).

"Signed-off By" is required as we are going to upstream this driver, so all contributions need to be appropriately tagged - see https://www.kernel.org/doc/html/latest/process/submitting-patches.html#sign-your-work-the-developer-s-certificate-of-origin. Please note "sorry, no pseudonyms or anonymous contributions".

@naushir We were talking about this change being needed earlier in the week - one fewer job for you then.

**Comment by naushir** [comment]
Once the warning are cleared, LGTM.

File: drivers/media/i2c/imx477.c [hunk 1]
```diff
@@ -25,6 +25,10 @@ static int dpc_enable = 1;
 module_param(dpc_enable, int, 0644);
 MODULE_PARM_DESC(dpc_enable, "Enable on-sensor DPC");
 
+static int trigger_mode = -1;
```
Comment (6by9) [inline drivers/media/i2c/imx477.c]: It was fine using a value of 0, but a static variable is automatically zeroed and so didn't need it explicitly.

```static int trigger_mode;```
would have been fine.

**Comment by neocortex-vision** [comment]
Thanks, done. Regarding initialisation, default (ineffective) value is now -1.

**Review by 6by9 — COMMENTED** [review]
Thanks for the update.

Could you also squash your commits together. We apply them exactly as presented in the PR, so don't need your development history.

**Comment by pelwell** [comment]
If the aim is to squash everything into one commit, especially if the commit message on the first commit doesn't require modification, then we can squash during the merge, but for more complicated combinations it's necessary for the submitter to squash before pushing.

**Comment by 6by9** [comment]
Upstream may point out that
```if (trigger_mode != 0)```
can be simplified to 
```if (trigger_mode)```
but otherwise that looks fine to me. Thanks for your efforts.

**Comment by neocortex-vision** [comment]
Thanks for the feedback and excuse the explicit initialisation/checking (it's habitual, coming from a MISRA background).
I was planning to add hsync, vsync and frstobe signal parameters but I see this may have already been planned @naushir.

**Pull request merged**